internal/packages/archetype/data_stream_manifest.go

```diff
@@ -27,4 +27,6 @@ streams:{{if eq .Manifest.Type "logs" }}
         title: Period
         default: 10s
 {{- end}}
+elasticsearch:
+  source_mode: synthetic
```

Comment:
Should it be included also `time_series` setting as synthetic `_source` is just GA for index.mode set to time_series ?
https://www.elastic.co/guide/en/elasticsearch/reference/8.7/mapping-source-field.html#synthetic-source

Reply:
This would look like a sane default, yes. But then I wonder if this should be enabled by default in logs data streams, not sure if we want to enable time series mode there.

Reply:
I've introduced a new question to ask for enabling time_series

```
 $ rm -rf data_stream/foo ; elastic-package create data-stream ; cat data_stream/foo/manifest.yml 
Create a new data stream
? Data stream name: foo
? Data stream title: Foo
? Type: metrics
? Enable synthetic source? Yes
? Enable time series? Yes
New data stream has been created: foo
Done
title: "Foo"
type: metrics
streams:
  - input: sample/metrics
    title: Sample metrics
    description: Collect sample metrics
    vars:
      - name: period
        type: text
        title: Period
        default: 10s
elasticsearch:
  source_mode: synthetic
  index_mode: time_series
```

This last question just is shown when the user selects Yes to "Enable synthetic source?" question.

So, the user could choose:
- not to use synthetic source
- use just synthetic source
- use both synthetic source and time_series

Maybe it's not needed and if so, I could revert the latest changes so this PR just introduces `source_mode` key. WDYT @jsoriano  ?

Reply:
Looks good, but maybe another option would be to make it more explicit, asking first if they want to enable time series and synthetic source, something like this:
```
? Enable time series and synthetic source? No
? Enable synthetic source? Yes
New data stream has been created: foo
```

If the first answer is affirmative, the second question is not done:
```
? Enable time series and synthetic source? Yes
New data stream has been created: foo
```

Reply:
Ok, I'll change it :+1: 
I'll add both question that by default they are true (yes answer).

Reply:
Examples:
- Just synthetic source:
```
Create a new data stream
? Data stream name: foo
? Data stream title: Foo
? Type: metrics
? Enable time series and synthetic source? No
? Enable synthetic source? Yes
New data stream has been created: foo
Done
title: "Foo"
type: metrics
streams:
  - input: sample/metrics
    title: Sample metrics
    description: Collect sample metrics
    vars:
      - name: period
        type: text
        title: Period
        default: 10s
elasticsearch:
  source_mode: synthetic
```
- Both synthetic and time_series:
```
Create a new data stream
? Data stream name: foo
? Data stream title: Foo
? Type: metrics
? Enable time series and synthetic source? Yes
New data stream has been created: foo
Done
title: "Foo"
type: metrics
streams:
  - input: sample/metrics
    title: Sample metrics
    description: Collect sample metrics
    vars:
      - name: period
        type: text
        title: Period
        default: 10s
elasticsearch:
  source_mode: synthetic
  index_mode: time_series
```